gril/grilreply.c

```diff
@@ -873,13 +888,16 @@ GSList *g_ril_reply_parse_get_calls(GRil *gril, const struct ril_msg *message)
 
 		ofono_call_init(call);
 		call->status = parcel_r_int32(&rilp);
-		call->id = parcel_r_int32(&rilp);
+		call->id = parcel_r_int32(&rilp) & 0xFF;
```

Comment:
Where does this 0xFF come?

Reply:
The [S3 Neo java file](https://github.com/LineageOS/android_device_samsung_s3ve3g/blob/ae9617a46f6899a3a5544d15e6ece879fbf7d66f/ril/HlteRIL.java#L149) says `& 0xff to truncate to 1 byte added for us, not in RIL.java`.  
Also in the [S7](https://github.com/Oliv4945/android_device_samsung_hero-common/blob/los-14.1-halium-7.1-ut/ril/telephony/java/com/android/internal/telephony/SlteRIL.java#L241) without the comment.

Reply:
Okay, good catch, but now you apply this unconditionally for all phones. You need again an if() block around that to only be valid for those Samsung plugins.

Reply:
If Android's RIL is on one byte it should not be armful but ok, done in 4c902e6d.
I made a mistake with my `git rebase` so this commit also reduce the number of lines used by `vendor == OFONO_RIL_VENDOR_SAMSUNG_EXYNOS_8890 || vendor == OFONO_RIL_VENDOR_SAMSUNG_MSM_822x`.

I believe that all your comments are addressed :)

Reply:
If Android's RIL is on one byte it should not be armful but ok, done in 4c902e6.
I made a mistake with my git rebase so this commit also reduce the number of lines used by vendor == OFONO_RIL_VENDOR_SAMSUNG_EXYNOS_8890 || vendor == OFONO_RIL_VENDOR_SAMSUNG_MSM_822x.

I believe that all your comments are addressed :)